### PR TITLE
[bitnami/prometheus-operator] Update the crds from upstream.

### PR DESCRIPTION
**Pull request by Cellebyte**
**Description of the change**
Updated the CRD to the version shipped with 0.38.0.

Most of the descriptions got an update and some minor changes to the APIs.
It has now thanosRuler support.
Also internall TLS configuration is now possible via the API.
They removed also Wrapper Objects from the API which are now standards from the
kubernetes-sigs.

[https://github.com/coreos/prometheus-operator/releases](https://github.com/coreos/prometheus-operator/releases)

**Benefits**

In the feature we can support thanosRuler rules Like prometheusrules.

**Possible drawbacks**
None (It looks like some basic improvements to the API nothing has changed from a User standpoint then adding some functionality to the API/CRD.

**Applicable issues**

  - fixes #
It could be that `helm upgrade` is not working.
I tested it with the former chart and the new chart in kind.
With the simple `values.yaml` the `helm upgrade` command is working as expected.

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

**Comment by ni-ajardan** [comment]
Seems like this also adds support for the disableCompaction option, which is required to be able to implement object storage sync for the Thanos sidecar.

Btw - this option is not documented in the Readme, nor is added to the prometheus.yaml template.

**Comment by Cellebyte** [comment]
@ni-ajardan should I add this to this PR? Then I will document it in the README and put it into the values and the prometheus.yaml template

**Comment by ni-ajardan** [comment]
> @ni-ajardan should I add this to this PR? Then I will document it in the README and put it into the values and the prometheus.yaml template

Would be very useful, if possible. I am actually surprised I am the first one to encounter this :)

**Comment by Cellebyte** [comment]
I think that regards to my currently 3 big PRs on this repo. Others need also time to get reviewed :) I will add the specific disableCompaction thing to the chart.

**Comment by ni-ajardan** [comment]
@Cellebyte you forgot to change the description in Readme :)

File: bitnami/prometheus-operator/crds/crd-alertmanager.yaml [hunk 1]
```diff
@@ -679,17 +695,19 @@ spec:
...
-                              description: Selects a key from a ConfigMap.
+                              description: Selects a key of a ConfigMap.
                               properties:
                                 key:
                                   description: The key to select.
                                   type: string
                                 name:
-                                  description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names'
+                                  description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+                                    TODO: Add other useful fields. apiVersion, kind,
```
Comment (miguelaeh) [inline bitnami/prometheus-operator/crds/crd-alertmanager.yaml]: I would not leave this comment here

Comment (Cellebyte) [reply bitnami/prometheus-operator/crds/crd-alertmanager.yaml]: @miguelaeh I used the following as a source.
[https://github.com/coreos/prometheus-operator/tree/release-0.38/example/prometheus-operator-crd](https://github.com/coreos/prometheus-operator/tree/release-0.38/example/prometheus-operator-crd)

**Comment by Cellebyte** [comment]
@miguelaeh is their something I could do? I can remove the `TODO` stuff from the CRDs if you want.

File: bitnami/prometheus-operator/crds/crd-alertmanager.yaml (hunk 1 above)
Comment (Cellebyte) [reply bitnami/prometheus-operator/crds/crd-alertmanager.yaml]: Used this regex `\s*TODO:[\w\s\.,\?]*'\n` --> `'\n` to replace every to do with nothing.

**Comment by miguelaeh** [comment]
Hi @Cellebyte ,
It seems that the PR passes our tests. Apart from that, what is the point about this:
>I am wondering if those new descriptions come from another source or something?

**Comment by Cellebyte** [comment]
> Hi @Cellebyte ,
> It seems that the PR passes our tests. Apart from that, what is the point about this:
> 
> > I am wondering if those new descriptions come from another source or something?

So they are from the official prometheus-operator tag 0.38 on github. I don't know why they changed so much. I don't know what the previous source was.

**Comment by ni-ajardan** [comment]
Since this PR is blocking me from moving forward on some issues, is there anything I can help with for this to be merged ? Thanks a lot for all the work !

**Comment by Cellebyte** [comment]
@ni-ajardan the current state is that it was tested and I think after resolving the fact with the CRD origin it will be merged. I don't know when @miguelaeh will schedule the PR. So it may be that it will take a couple of days until it gets resolved and merged or I get feedback of what todo here.

**Comment by miguelaeh** [comment]
Let's wait for the tags to be updated

**Comment by miguelaeh** [comment]
Hi again,
we are having an issue in our internal CI/CD because the name of the remote branch is the same as the name of a directory in the repository. We are fixing it, but just to clarify that is why @carrodher commited a test.
Thanks for the patient.

**Comment by Cellebyte** [comment]
@miguelaeh thanks for the information. Don't want to trigger a bug in your system ^^

**Comment by carrodher** [comment]
The opposite, thanks for bringing it up 😄

**Comment by miguelaeh** [comment]
We already solved it and the tags have been updated, so merging the PR.
Thank you for the patient.

**Pull request merged**